runtime/bcutil/defineclass.c

```diff
@@ -861,4 +885,81 @@ classCouldPossiblyBeShared(J9VMThread * vmThread, J9LoadROMClassData * loadData)
 	return ((0 != (loadData->classLoader->flags & J9CLASSLOADER_SHARED_CLASSES_ENABLED)) && !j9shr_Query_IsCacheFull(vm));
 }
 
+/* Return TRUE if anonClass and hostClass have the same package name.
+ * If anonymous class has no package name, then consider it to be part
+ * of host class's package. Return TRUE if anonymous class has no
+ * package name. Otherwise, return FALSE.
+ */
+static BOOLEAN
+hasSamePackageName(J9ROMClass *anonROMClass, J9ROMClass *hostROMClass) {
+	BOOLEAN rc = FALSE;
+
+	const U_8 *anonClassName = J9UTF8_DATA(J9ROMCLASS_CLASSNAME(anonROMClass));
+	const UDATA anonClassPackageNameLength = packageNameLength(anonROMClass);
+
+	const U_8 *hostClassName = J9UTF8_DATA(J9ROMCLASS_CLASSNAME(hostROMClass));
+	const UDATA hostClassPackageNameLength = packageNameLength(hostROMClass);
+
+	if ((0 == anonClassPackageNameLength)
```

Comment:
Host class name/length are only required when `anonClassPackageNameLength` is `0`.

Reply:
improved.
```
static BOOLEAN
hasSamePackageName(J9ROMClass *anonROMClass, J9ROMClass *hostROMClass) {
	BOOLEAN rc = FALSE;
	const UDATA anonClassPackageNameLength = packageNameLength(anonROMClass);

	if (0 == anonClassPackageNameLength) {
		rc = TRUE;
	} else {
		const U_8 *anonClassName = J9UTF8_DATA(J9ROMCLASS_CLASSNAME(anonROMClass));
		const U_8 *hostClassName = J9UTF8_DATA(J9ROMCLASS_CLASSNAME(hostROMClass));
		const UDATA hostClassPackageNameLength = packageNameLength(hostROMClass);
		if (J9UTF8_DATA_EQUALS(anonClassName, anonClassPackageNameLength, hostClassName, hostClassPackageNameLength)) {
			rc = TRUE;
		}
	}

	return rc;
}
```